README.md

```diff
@@ -7,341 +7,171 @@ This library provides debugging functionality to Ruby (MRI) 2.6 and later.
-
-```shell
-(rdbg) break 3                         # set breakpoint at line 3
-#0  BP - Line  /mnt/c/ko1/src/rb/ruby-debug/target.rb:3 (line)
+# Usage
 
-(rdbg) b 5                             # set breakpoint at line 5
-#1  BP - Line  /mnt/c/ko1/src/rb/ruby-debug/target.rb:5 (line)
+The debugger is designed to support a wide range of use cases, so you have many ways to use it.
 
-(rdbg) break                           # show all registered breakpoints
-#0  BP - Line  /mnt/c/ko1/src/rb/ruby-debug/target.rb:3 (line)
-#1  BP - Line  /mnt/c/ko1/src/rb/ruby-debug/target.rb:5 (line)
-```
+But essentially, it consists of 4 steps:
 
-You can see that two breakpoints are registered. Let's continue the program by `continue` command.
+1. [Activate the debugger in your program](#activate-the-debugger-in-your-program)
+1. Set breakpoints
+    - Through [`binding.break`](#the-bindingbreak-method)
+    - [Breakpoint commands](#breakpoint)
+1. Execute/continue your program and wait for it to hit the breakpoints
+1. Start debugging
+    - Here's the [full commands list](#console-commands)
+    - You can also type `help` or `help <command>` in the console to see commands
 
-```shell
-(rdbg) continue
-[1, 7] in target.rb
-      1| a = 1
-      2| b = 2
-=>    3| c = 3
-      4| d = 4
-      5| p [a, b, c, d]
-      6|
-      7| __END__
-=>#0    <main> at target.rb:3
+> **NOTE**
+> If you want to use remote console or VSCode/Chrome integration, the steps will be slightly different. Please also check the [remote debugging guide](/docs/remote_debugging.md) as well.
 
-Stop by #0  BP - Line  /mnt/c/ko1/src/rb/ruby-debug/target.rb:3 (line)
+## Common Usges
 
-(rdbg)
-```
+Here are the 2 most common usages of the debugger:
 
-You can see that we can stop at line 3.
-Let's see the local variables with `info` command, and continue.
-You can also confirm that the program will suspend at line 5 and you can use `info` command again.
-
-```shell
-(rdbg) info
-=>#0    <main> at target.rb:3
-%self => main
-a => 1
-b => 2
-c => nil
-d => nil
-
-(rdbg) continue
-[1, 7] in target.rb
-      1| a = 1
-      2| b = 2
-      3| c = 3
-      4| d = 4
-=>    5| p [a, b, c, d]
-      6|
-      7| __END__
-=>#0    <main> at target.rb:5
-
-Stop by #1  BP - Line  /mnt/c/ko1/src/rb/ruby-debug/target.rb:5 (line)
-
-(rdbg) info
-=>#0    <main> at target.rb:5
-%self => main
-a => 1
-b => 2
-c => 3
-d => 4
-
-(rdbg) continue
-[1, 2, 3, 4]
-```
+### Quick start (similar to `byebug` or `pry` use cases)
+  1.
+      ```rb
+      require "debug"
+      ```
+  1.
+      ```rb
+      # somewhere in your program
+      def target_method
+        binding.break
+      end
+      ```
+  1. When the program executes `target_method`, debugger will stop it and open up a console
 
-By the way, using `rdbg` command you can suspend your application with `C-c` (SIGINT) and enter the debug console.
-It will help that if you want to know what the program is doing.
+### Traditional debugger start
```

Comment:
Maybe instead of calling one a "Quick start" and the other "Traditional", we should just describe what they are ("Beakpoint in code" and "Command line").

Reply:
Yeah I'm struggling on these titles. The goal is to let the first part catching eye balls fairly quickly, so I chose `Quick start`.
But then I'm not sure how to name the second one.